test/module/shared_model/backend_proto/common_objects/proto_common_objects_factory_test.cpp

```diff
@@ -35,7 +35,7 @@ class PeerTest : public ProtoFixture {
  * @then peer is successfully initialized
  */
 TEST_F(PeerTest, ValidPeerInitialization) {
-  auto peer = factory.createPeer(valid_address, valid_pubkey);
+  auto peer = factory.createPeer(valid_address, valid_pubkey, std::string(""));
```

Comment:
we should test two cases here: peer having and not having a certificate. for the first case, a cert should better be some non-empty string defined like `valid_address`, and in both cases the check should be done below (in `peer.match`)